components/scream/src/mct_coupling/scream_scorpio_interface.cpp

```diff
@@ -12,15 +12,11 @@ using scream::Int;
 extern "C" {
 
 // Fortran routines to be called from C++
-  void register_infile_c(const std::string (&filename));
-  void grid_read_data_array_c_real_1d(const char*&& filename, const char*&& varname, const Int dim1_length, Real *hbuf);
-  void grid_read_data_array_c_real_2d(const char*&& filename, const char*&& varname, const Int dim1_length, const Int dim2_length, Real *hbuf);
-  void grid_read_data_array_c_real_3d(const char*&& filename, const char*&& varname, const Int dim1_length, const Int dim2_length, const Int dim3_length, Real *hbuf);
-  void grid_read_data_array_c_real_4d(const char*&& filename, const char*&& varname, const Int dim1_length, const Int dim2_length, const Int dim3_length, const Int dim4_length, Real *hbuf);
-  void grid_read_data_array_c_int_1d(const char*&& filename, const char*&& varname, const Int dim1_length, Int *hbuf);
-  void grid_read_data_array_c_int_2d(const char*&& filename, const char*&& varname, const Int dim1_length, const Int dim2_length, Int *hbuf);
-  void grid_read_data_array_c_int_3d(const char*&& filename, const char*&& varname, const Int dim1_length, const Int dim2_length, const Int dim3_length, Int *hbuf);
-  void grid_read_data_array_c_int_4d(const char*&& filename, const char*&& varname, const Int dim1_length, const Int dim2_length, const Int dim3_length, const Int dim4_length, Int *hbuf);
+  void register_infile_c(const char*&& filename);
```

Comment:
While you're at it, we could start using better names for our interoperable functions. When I see `_c` at the end of a function name, I think that the impl of that fcn is in C, while here is the other way around (based on the commend above).

Someone (maybe Jeff) proposed a better suffix: if foo is impl-ed in F but meant to be called from C, then name it `foo_c2f`, and viceversa. This underlines the code stack flow pattern: you start from c and go to f.

Reply:
Note: I'm not asking you to change all existing fcns, but the ones we write anew, we could already use a more descriptive suffix.

Reply:
Yes, you are correct.  I've made the change to all routines here.  Better to start with the proper way rather than promise to be better in the future.